include/zephyr/rtio/rtio_executor_inplace.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_RTIO_RTIO_EXECUTOR_INPLACE_H_
+#define ZEPHYR_INCLUDE_RTIO_RTIO_EXECUTOR_INPLACE_H_
+
+#include <rtio/rtio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief RTIO API
+ * @defgroup rtio_api RTIO API
+ * @ingroup rtio
+ * @{
+ */
+
+
+/**
+ * @brief Execte Inplace
+ *
+ * Iterates over the submission queue executing each requested transfer
+ * as needed. Different operation chains are done in order and are not done
+ * in parallel using threads or DMA. Each rtio_iodev may still be asynchronous
+ * and perhaps even use the platforms DMA, but no specifics are detailed here
+ * other than the device is asked to perform an IO operation with a buffer.
```

Comment:
Is priority based ordering used in this or just simple FIFO?

Reply:
priority is given and used when DMA is to be used. Potentially it might be used in the future in other ways, but predominantly I've added because DMA controllers typically let you set priorities that get used by the hardware to prioritize things.